packages/core/src/index.ts

```diff
@@ -0,0 +1,191 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/require-await */
+
+export enum RouteType {
+  GMP = "gmp",
+  FUNGIBLE = "fungible",
+}
+
+export enum Environment {
+  LOCAL = "local",
+  DEVNET = "devnet",
+  TESTNET = "testnet",
+  MAINNET = "mainnet",
+}
+
+export enum FeeHandlerType {
+  BASIC = "basic",
+  PERCENTAGE = "percentage",
+}
+
+export enum Network {
+  EVM = "evm",
+  SUBSTRATE = "substrate",
+}
+
+export enum SecurityModel {
+  MPC = "mpc",
+}
+
+export type Domain = {
+  sygmaId: number;
+  // https://chainagnostic.org/CAIPs/caip-2
+  caipId: string;
+  chainId: number;
+  name: string;
+  iconUrl?: string;
+  type: Network;
+};
+
+export type Resource = EvmResource | SubstrateResource;
+
+export enum ResourceType {
+  FUNGIBLE = "fungible",
+  NON_FUNGIBLE = "nonfungible",
+  PERMISSIONED_GENERIC = "permissionedGeneric",
+  PERMISSIONLESS_GENERIC = "permissionlessGeneric",
+}
+
+interface BaseResource {
+  sygmaResourceId: string;
+  // https://chainagnostic.org/CAIPs/caip-19
+  caip19: string;
+  type: ResourceType;
+  iconUrl?: string;
+  native?: boolean;
+  burnable?: boolean;
+  symbol?: string;
+  decimals?: number;
+}
+
+export type EvmResource = BaseResource & {
+  address: string;
+};
+
+export type XcmMultiAssetIdType = {
+  concrete: {
+    parents: number;
+    interior:
+      | "here"
+      | {
+          x3: Array<{ parachain: number } | { generalKey: [number, string] }>; // This is a tuple of length and value
+        };
+  };
+};
+
+export type SubstrateResource = BaseResource & {
+  assetID?: number;
+  assetName: string;
+  xcmMultiAssetId: XcmMultiAssetIdType;
+};
+
+export type Route = {
+  fromDomain: Domain;
+  toDomain: Domain;
+  resource: Resource;
+  feeHandler: FeeHandler;
+};
+
+export type TransferStatus = "pending" | "executed" | "failed";
+
+export type TransferStatusResponse = {
+  status: TransferStatus;
+  explorerUrl: string;
+  sourceHash: string;
+  destinationHash: string;
+  depositNonce: number;
+  fromDomainId: number;
+  toDomainId: number;
+};
+
+export interface BaseConfig<T> {
+  sygmaId: number;
+  chainId: number;
+  caipId: number;
```

Comment:
this needs to be a string IMO

Reply:
For sure as caipID is string by standard definition, not a number